cranelift/codegen/src/dominator_tree.rs

```diff
@@ -482,7 +482,42 @@ impl DominatorTreePreorder {
     pub fn dominates(&self, a: Block, b: Block) -> bool {
         let na = &self.nodes[a];
         let nb = &self.nodes[b];
-        na.pre_number <= nb.pre_number && na.pre_max >= nb.pre_max
+        // debug_assert!(na.pre_number != 0);
+        // debug_assert!(nb.pre_number != 0);
+        if a == b {
+            // check first if they're the same block
+            // if they are then they dominate themselves
+            true
+        } else if nb.pre_number == 0 {
+            // if the supposedly dominated is not reachable (pre_number = 0), is it
+            // really being dominated though ?
+            false
+        } else {
+            na.pre_number <= nb.pre_number && na.pre_max >= nb.pre_max
+        }
+    }
+
+    /// Checks if one instruction dominates another.
+    ///
+    /// From @jameysharp:
+    /// Instruction a dominates instruction b if either of these conditions are true:
+    ///    a and b are in the same block, and a is not after b.
+    ///    Or the block containing a dominates the block containing b.
+    ///
+    ///    "I would prefer to have dominates_inst next to the dominates method that it uses.
+    ///     Then someone looking at one of them can easily see the other one too."
+    ///
```

Comment:
Could you rephrase this comment from Jamey as a comment about the function implementation?

Reply:
resolved